### ISSUE TITLE
Bad geoip lookup returns invalid hash

### ISSUE THREAD

**Issue by tomsommer** [logstash-plugins/logstash-filter-geoip#50]
Please see https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/298#issuecomment-156066999

**Comment by guyboertje** [comment]
@tomsommer - in the test for this plugin we do verify that lookups on bad IP values return sane values.  These tests pass. A bad IP lookup returns an empty hash.  The test setup (config, Geoip DB etc.), though, may not be the same as your setup.

Please email or post your geoip filter config section.

**Comment by tomsommer** [comment]
```
 geoip {
                        database => "/usr/share/GeoIP/GeoIPCity.dat"
                        source => "dstip"
                        target => "dstip_geoip"
                }
```

**Comment by guyboertje** [comment]
@tomsommer - I have found the bug and it only manifests when LS is set to log to File.

I will submit a fix today.

**Comment by tomsommer** [comment]
Awesome

**Comment by guyboertje** [comment]
PR #52

**Comment by sbaker-gre** [comment]
I'm hitting this issue in my first logstash setup and not sure how to get troubleshoot my logstash configuration or get past it. I'm just getting familiar with logstash and ruby gems. Is the fix for this in the most current release? I just reinstalled logstash today.

**Comment by guyboertje** [comment]
@sbaker-gre - if you are using LS 2.1.1 you will see in the file called `Gemfile.jruby-1.9.lock` this `logstash-filter-geoip (2.0.5)`.
This bug was fixed in logstash-filter-geoip 2.0.3

I suggest that you head over to [our discussion forum](https://discuss.elastic.co/c/logstash) to get help with investigating what is going on.  You will need to explain your setup and what the actual problem is.
